.github/workflows/requirements.yml

```diff
@@ -0,0 +1,74 @@
+# Copyright 2021 Agnostiq Inc.
+#
+# This file is part of Covalent.
+#
+# Licensed under the GNU Affero General Public License 3.0 (the "License").
+# A copy of the License may be obtained with this software package or at
+#
+#      https://www.gnu.org/licenses/agpl-3.0.en.html
+#
+# Use of this file is prohibited except in compliance with the License. Any
+# modifications or derivative works of this file must retain this copyright
+# notice, and modified files must contain a notice indicating that they have
+# been altered from the originals.
+#
+# Covalent is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE. See the License for more details.
+#
+# Relief from the License may be granted by purchasing a commercial license.
+
+name: requirements
+
+on: [pull_request]
+
+jobs:
+  requirements:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out head
+        uses: actions/checkout@v3
+        with:
+          persist-credentials: false
+
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.8
+
+      - name: Install covalent
+        run: python -m pip install .
+
+      - name: Install packages
+        run: python -m pip install pip-check-reqs
+
+      - name: Check extra core requirements
+        run: pip-extra-reqs -r werkzeug covalent covalent_dispatcher covalent_ui
+
+      - name: Check missing SDK requirements
+        run: >
+          pip-missing-reqs
+          --ignore-module=covalent_ui.*
+          --ignore-module=covalent.*
+          --ignore-module=urllib3.*
+          --ignore-module=pkg_resources
+          covalent
+
+      - name: Check missing dispatcher requirements
+        run: >
+          pip-missing-reqs
+          --ignore-module=covalent_ui.*
+          --ignore-module=covalent.*
+          --ignore-module=covalent_dispatcher
+          --ignore-module=distributed.*
+          --ignore-module=click
```

Comment:
should `click` be added to the requirements?

Reply:
click is included with distributed, so I guess we don't need it in requirements?

Reply:
```
distributed==2022.6.0
  - click [required: >=6.6, installed: 8.1.3]
  - cloudpickle [required: >=1.5.0, installed: 2.0.0]
  - dask [required: ==2022.6.0, installed: 2022.6.0]
    - cloudpickle [required: >=1.1.1, installed: 2.0.0]
    - fsspec [required: >=0.6.0, installed: 2022.8.1]
    - packaging [required: >=20.0, installed: 21.3]
      - pyparsing [required: >=2.0.2,!=3.0.5, installed: 2.4.7]
    - partd [required: >=0.3.10, installed: 1.3.0]
      - locket [required: Any, installed: 1.0.0]
      - toolz [required: Any, installed: 0.12.0]
    - pyyaml [required: >=5.3.1, installed: 6.0]
    - toolz [required: >=0.8.2, installed: 0.12.0]
  - jinja2 [required: Any, installed: 3.1.2]
    - MarkupSafe [required: >=2.0, installed: 2.1.1]
  - locket [required: >=1.0.0, installed: 1.0.0]
  - msgpack [required: >=0.6.0, installed: 1.0.4]
  - packaging [required: >=20.0, installed: 21.3]
    - pyparsing [required: >=2.0.2,!=3.0.5, installed: 2.4.7]
  - psutil [required: >=5.0, installed: 5.9.0]
  - pyyaml [required: Any, installed: 6.0]
  - sortedcontainers [required: !=2.0.1,!=2.0.0, installed: 2.4.0]
  - tblib [required: >=1.6.0, installed: 1.7.0]
  - toolz [required: >=0.8.2, installed: 0.12.0]
  - tornado [required: >=6.0.3, installed: 6.2]
  - urllib3 [required: Any, installed: 1.26.12]
  - zict [required: >=0.1.3, installed: 2.2.0]
    - heapdict [required: Any, installed: 1.0.1]
```

Reply:
the purpose of `pip-missing-reqs` is to force us to require anything explicitly imported, e.g., in case we stop using distributed and remove it then it won't break things